common/djangoapps/third_party_auth/views.py

```diff
@@ -36,7 +37,7 @@ def saml_metadata_view(request):
     Get the Service Provider metadata for this edx-platform instance.
     You must send this XML to any Shibboleth Identity Provider that you wish to use.
     """
-    if not SAMLConfiguration.is_enabled():
+    if not SAMLConfiguration.is_enabled(get_current_site_default()):
```

Comment:
No need to use `get_current_site_default` here. You can use the request object to get the current site, `request.site`.